src/rust/engine/process_execution/src/remote_cache.rs

```diff
@@ -512,23 +517,37 @@ impl crate::CommandRunner for CommandRunner {
     };
 
     if result.exit_code == 0 && self.cache_write {
-      let write_result = self
-        .update_action_cache(
-          &context,
-          &request,
-          &result,
-          &self.metadata,
-          &command,
-          action_digest,
-          command_digest,
-        )
-        .await;
-      if let Err(err) = write_result {
-        log::warn!("Failed to write to remote cache: {}", err);
-        context
-          .workunit_store
-          .increment_counter(Metric::RemoteCacheWriteErrors, 1);
-      };
+      context
+        .workunit_store
+        .increment_counter(Metric::RemoteCacheWriteStarted, 1);
+      let command_runner = self.clone();
+      let result = result.clone();
+      // NB: We use `TaskExecutor::spawn` instead of `tokio::spawn` to ensure logging still works.
+      let _write_join = self.executor.spawn(
+        async move {
+          let write_result = command_runner
+            .update_action_cache(
+              &context,
+              &request,
+              &result,
+              &command_runner.metadata,
+              &command,
+              action_digest,
+              command_digest,
+            )
+            .await;
+          context
+            .workunit_store
+            .increment_counter(Metric::RemoteCacheWriteFinished, 1);
+          if let Err(err) = write_result {
+            log::warn!("Failed to write to remote cache: {}", err);
+            context
+              .workunit_store
+              .increment_counter(Metric::RemoteCacheWriteErrors, 1);
```

Comment:
Should not block this PR, but we should make sure cache errors still show up in stats when increments are "backgrounded" like this.